cpp/src/arrow/acero/hash_join_benchmark.cc

```diff
@@ -178,25 +180,30 @@ class JoinBenchmark {
     scheduler_->RegisterEnd();
 
     DCHECK_OK(scheduler_->StartScheduling(
-        0 /*thread index*/, std::move(schedule_callback),
-        static_cast<int>(2 * settings.num_threads) /*concurrent tasks*/,
-        settings.num_threads == 1));
+        /*thread_id=*/0,
+        [&](std::function<Status(size_t)> task) -> Status {
+          return thread_pool_->Spawn([&, task]() { DCHECK_OK(task(thread_indexer_())); });
```

Comment:
Why did you choose `Spawn()` + conditional variable not `Submit()` + `Future::status()` (or `Spawn()` + `ThreadPool::WaitForIdle()`)? Is it easy to write/maintain?

Reply:
Updated.